mp/src/game/client/neo/ui/neo_hud_ammo.cpp

```diff
@@ -137,169 +123,165 @@ void CNEOHud_Ammo::DrawAmmo() const
-		auto maxBulletsWeCanDisplay = (maxSpaceAvaliableForBullets / bulletWidth);
-		auto maxBulletsWeCanDisplayWithPlus = ((maxSpaceAvaliableForBullets - plusWidth) / bulletWidth);
-		auto bulletsOverflowing = maxBulletsWeCanDisplay < magSizeMax;
+	const int maxSpaceAvaliableForBullets = digit_max_width;
+	const int bulletWidth = surface()->GetCharacterWidth(m_hBulletFont, *ammoChar);
+	const int plusWidth = surface()->GetCharacterWidth(m_hBulletFont, '+');
+	const int maxBulletsWeCanDisplay = (maxSpaceAvaliableForBullets / bulletWidth);
+	const int maxBulletsWeCanDisplayWithPlus = ((maxSpaceAvaliableForBullets - plusWidth) / bulletWidth);
+	const int bulletsOverflowing = maxBulletsWeCanDisplay < magSizeMax;
 
-		if(bulletsOverflowing)
-		{
-			magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
-		}
+	if(bulletsOverflowing)
+	{
+		magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
+	}
 
-		magSizeMax = min(magSizeMax, 64);
-		char bullets[64]{ '\0' };
-		for(int i = 0; i < magSizeMax; i++)
-		{
-			bullets[i] = *ammoChar;
-		}
+	magSizeMax = min(magSizeMax, 100);
+	char bullets[100]{ '\0' };
+	for(int i = 0; i < magSizeMax; i++)
+	{
+		bullets[i] = *ammoChar;
+	}
 
-		int magAmountToDrawFilled = magSizeCurrent;
+	int magAmountToDrawFilled = magSizeCurrent;
 		
-		if(bulletsOverflowing)
+	if(bulletsOverflowing)
+	{
+		bullets[magSizeMax - 1] = '+';
+
+		if(maxClip == magSizeCurrent)
 		{
-			bullets[magSizeMax - 1] = '+';
-
-			if(maxClip == magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax;
-			} else if(magSizeMax - 1 < magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax - 1;
-			} else
-			{
-				magAmountToDrawFilled = magSizeCurrent;
-			}
+			magAmountToDrawFilled = magSizeMax;
+		} else if(magSizeMax - 1 < magSizeCurrent)
+		{
+			magAmountToDrawFilled = magSizeMax - 1;
+		} else
+		{
+			magAmountToDrawFilled = magSizeCurrent;
 		}
+	}
 		
-		wchar_t unicodeBullets[64];
-		g_pVGuiLocalize->ConvertANSIToUnicode(bullets, unicodeBullets, sizeof(unicodeBullets));
+	wchar_t unicodeBullets[100];
+	g_pVGuiLocalize->ConvertANSIToUnicode(bullets, unicodeBullets, sizeof(unicodeBullets));
 		
-		if (magAmountToDrawFilled > 0)
-		{
-			surface()->DrawSetTextFont(m_hBulletFont);
-			surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
-			surface()->DrawPrintText(unicodeBullets, magAmountToDrawFilled);
-		}
+	if (magAmountToDrawFilled > 0)
+	{
+		surface()->DrawSetTextFont(m_hBulletFont);
+		surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
+		surface()->DrawPrintText(unicodeBullets, magAmountToDrawFilled);
+	}
 
-		if(maxClip > 0)
-		{
-			if (magSizeMax > 0) {
-				surface()->DrawSetTextColor(textColorTransparent);
-				surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
-				surface()->DrawPrintText(unicodeBullets, magSizeMax);
-			}
+	if(maxClip > 0)
+	{
+		if (magSizeMax > 0) {
+			const wchar_t* firstEmptyRound = unicodeBullets + magAmountToDrawFilled;
```

Comment:
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/f80f1120-24d4-406e-a9d1-d93e55055044)

I was trying to avoid converting ANSI to unicode twice, so yes I wanted to get a pointer to the middle of the uncodeBullets wchar_t array. Multiplying magAmountToDrawFilled by the size of wchar_t was resulting in a bunch of the uninitialised character.

In retrospect I could just pass the pointer to the first bullet in unicodeBullets, since we are already preventing bullets being drawn one on top of another by moving the xpos of the second drawsettextpost, but this could cause problems if we ever in the future implement mixed bullet types in the same mag or something.

Reply:
Ok, so apparently array+int is essentially &array[int] regardless of the array's type width. This should be fine then.